crates/scroll/cli/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+impl Cli {
+    /// Parsers only the default CLI arguments
+    pub fn parse_args() -> Self {
+        Self::parse()
+    }
+
+    /// Parsers only the default CLI arguments from the given iterator
+    pub fn try_parse_args_from<I, T>(itr: I) -> Result<Self, clap::error::Error>
+    where
+        I: IntoIterator<Item = T>,
+        T: Into<OsString> + Clone,
+    {
+        Self::try_parse_from(itr)
+    }
+}
+
+impl<C, Ext> Cli<C, Ext>
+where
+    C: ChainSpecParser<ChainSpec = ScrollChainSpecParser>,
+    Ext: clap::Args + fmt::Debug,
+{
+    /// Execute the configured cli command.
+    ///
+    /// This accepts a closure that is used to launch the node via the
+    /// [`NodeCommand`](reth_cli_commands::node::NodeCommand).
+    pub fn run<L, Fut>(mut self, launcher: L) -> eyre::Result<()>
+    where
+        L: FnOnce(WithLaunchContext<NodeBuilder<Arc<DatabaseEnv>, C::ChainSpec>>, Ext) -> Fut,
+        Fut: Future<Output = eyre::Result<()>>,
+    {
+        // add network name to logs dir
+        self.logs.log_file_directory =
+            self.logs.log_file_directory.join(self.chain.chain().to_string());
+
+        let _guard = self.init_tracing()?;
+        info!(target: "reth::cli", "Initialized tracing, debug log directory: {}", self.logs.log_file_directory);
+
+        // Install the prometheus recorder to be sure to record all metrics
+        let _ = install_prometheus_recorder();
+
+        let runner = CliRunner::default();
+        match self.command {
+            Commands::Node(command) => {
+                runner.run_command_until_exit(|ctx| command.execute(ctx, launcher))
+            }
+            Commands::Init(command) => {
+                runner.run_blocking_until_ctrl_c(command.execute::<ScrollNode>())
+            }
+            Commands::DumpGenesis(command) => runner.run_blocking_until_ctrl_c(command.execute()),
+            Commands::Db(command) => {
+                runner.run_blocking_until_ctrl_c(command.execute::<ScrollNode>())
+            }
+            Commands::Stage(command) => runner.run_command_until_exit(|ctx| {
+                command.execute::<ScrollNode, _, _>(ctx, ScrollExecutorProvider::scroll)
+            }),
+            Commands::P2P(command) => runner.run_until_ctrl_c(command.execute()),
+            Commands::Config(command) => runner.run_until_ctrl_c(command.execute()),
+            Commands::Recover(command) => {
+                runner.run_command_until_exit(|ctx| command.execute::<ScrollNode>(ctx))
+            }
+            Commands::Prune(command) => runner.run_until_ctrl_c(command.execute::<ScrollNode>()),
+        }
+    }
+
+    /// Initializes tracing with the configured options.
+    ///
+    /// If file logging is enabled, this function returns a guard that must be kept alive to ensure
+    /// that all logs are flushed to disk.
+    pub fn init_tracing(&self) -> eyre::Result<Option<FileWorkerGuard>> {
+        let guard = self.logs.init_tracing()?;
+        Ok(guard)
+    }
+}
```

Comment:
question: Had you considered using the `Cli` type from the `reth` crate?

Reply:
I see. Let's leave it as is for now and consider whether we should open an upstream PR to make it generic over the Node. However, we may diverge from the CLI and commands for native Ethereum as we may have L2-specific functionality that we need to introduce.

Reply:
understood, it could be interesting to have someway to define extensions for the commands instead of having to redefine the whole enum. Maybe something to look into.